yt/utilities/lib/particle_mesh_operations.pyx

```diff
@@ -87,6 +87,7 @@ def CICDeposit_2(np.float64_t[:] posx,
                  np.float64_t[:] mass,
                  np.int64_t npositions,
                  np.float64_t[:, :] field,
+                 np.int_t[:, :] field_mask,
```

Comment:
Can you change this, and the previous declaration, to be an int with an explicit size?  Since you're just setting it to 1, you could use `uint8` (which I think is the backing store for `bool`)